fiboa_cli/publish.py

```diff
@@ -0,0 +1,378 @@
+        super().__init__()
+        self.cmd_title = f"Convert {dataset}"
+        self.dataset = dataset
+        self.data_url = data_url or f"{self.url_base}/{self.dataset}"
+        self.s3_upload_path = (
+            s3_upload_path or f"s3://us-west-2.opendata.source.coop/fiboa/data/{self.dataset}/"
+        )
+
+        if ConvertData.converters.is_converter(self.dataset):
+            raise Exception(f"'{self.dataset}' is not a converter")
+        try:
+            self.converter = ConvertData.converters.load(self.dataset)
+        except (ImportError, NameError, OSError, RuntimeError, SyntaxError) as e:
+            raise Exception(f"Converter for '{self.dataset}' not available or faulty: {e}") from e
+
+    def exc(self, cmd):
+        assert os.system(cmd) == 0
+
+    def check_command(self, cmd, name=None):
+        if os.system(f"{cmd} --version") != 0:
+            self.error(f"Missing command {cmd}. Please install {name or cmd}")
+            sys.exit(1)
+
+    def download_data_survey(self, base):
+        data_survey = (
+            os.getenv("FIBOA_DATA_SURVEY")
+            or f"https://raw.githubusercontent.com/fiboa/data-survey/refs/heads/main/data/{base}.md"
+        )
+        response = requests.get(data_survey)
+        if not response.ok:
+            self.warning(
+                f"Missing data survey {base}.md at {data_survey}. Can not auto-generate file"
+            )
+        else:
+            return response.text
+
+    @cache
+    def get_data_survey(self):
+        base = self.dataset.replace("_", "-").upper()
+        text = self.download_data_survey(base)
+        mapping = {
+            "data provider (legal entity)": "provider",
+            "submitter (affiliation)": "submitter",
+        }
+        # override data survey location with env variable, e.g. for unmerged pull-requests
+        data = {
+            "provider": self.converter.provider,
+            "license": self.converter.license,
+            "projection": "",
+            "homepage": "",
+            "submitter": "Fiboa project",
+        }
+        properties = {}
+        if text:
+            data.update({a.lower(): b for a, b in re.findall(r"- \*\*(.+?):\*\* (.+?)\n", text)})
+            properties = {
+                mapping.get(a.lower(), a.lower()): b.strip()
+                for a, b in re.findall(r"\n\|\s*(\w+)[^|]*\|[^|]*\|[^|]*\|([^|]*)\|", text)
+            }
+
+        return data, properties
+
+    def readme_attribute_table(self, stac_data, properties):
+        def description(name):
+            m = self.converter.columns
+            reverse = dict(zip(m.values(), m.keys()))
+            return (
+                properties.get(reverse.get(name))
+                or properties.get(name)
+                or DESCRIPTIONS.get(name, "")
+            )
+
+        cols = [["Property", "**Data Type**", "Description"]] + [
+            [
+                s["name"],
+                re.search(r"\w+", s["type"])[0],
```

Comment:
Are there any instancec where the datatype coud be something like `[string]`, `Array<string>`, or `Map<stirng, int>` or so?